Quantum-Control-Applications/ESR/Use case 1 - Sekhar Lab - CPMG/README.md

```diff
@@ -0,0 +1,102 @@
+# Carr-Purcell-Meiboom-Gill (CPMG) in an NV ensemble with electron spin resonance (ESR)
+
+_author: Kevin Villegas Rosales_
+
+_Demonstrated in the Lab of Prof. Sekhar at Darmouth College https://sites.dartmouth.edu/quantum-spin-lab/._
+
+_Demonstrated on the experiment of Ethan Williams. After two days of unboxing the OPX we
+were able to obtain the results listed below._
+
+_Important note: The code in this folder is the exact code that we used
+to run the experiment, so it is tailored for the specific software environment
+and set up. Thus, this is code is for inspirational purposes._
+
+## The goal
+
+To obtain the decoherence time of an ensemble of NV centers with CPMG dynamical decoupling
+sequence. Thus, it is a starting point for performing future noise spectroscopy experiments.
+
+## The device
+
+The device is a crystal containing an NV ensemble. The crystal is embedded in a room
+temperature ESR setup with a resonator at 2.5 GHz and 13 mT applied magnetic field.
+Note that the amplitude of the echo signal was obtained during the run-time
+of the experiment thanks for the demodulation capabilities of the OPX. Thus, there
+was no need for post-processing and the results shown below were obtained immediately
+after the end of the measurement.
+
+## Methods and results
+
+### CPMG sequence
+
+CPMG is a dynamical decoupling sequence that consist of a train of pulses. The effect of this
+train of pulses is to isolate the NV ensemble from noise at certain frequencies. Thus, it can
+serve to extend the lifetime of information encoded in qubits. A CPMG sequence can also be used to
+probe the noise environment surrounding the spins [1]. The image below displays the
+train of pulses applied in this experiment. After the hyperpolarization of spins, an initial
+`X_pi/2` pulse is sent and it is followed by a train of `Y_pi`. The delay between two consecutive
+`Y_pi` pulses is `2*tau`. In ESR experiments the echo amplitude is detected at `tau` time after
+`Y_pi` pulses.
+
+![sequence](CPMG_seq.png)
+
+### Echo amplitude decay in CPMG
+
+The measured decay traces with CPMG are shown below. The `y-axis` list the echo amplitude as demodulated
```

Comment:
How many pulses were there? Was it a constant number or did it change when `tau` was changed?